### PR TITLE
CXP-1255: remove pagination on locales filter

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/front/src/hooks/useLocales.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import {useQuery} from 'react-query';
-import {Locale} from '../models/Locale';
+import {Locale} from '../components/ProductSelection/models/Locale';
```
Comment (masacc) [inline components/catalogs/front/src/hooks/useLocales.ts:2]: you should not move this `useLocales` hook in this directory for 2 reasons:
- it uses a model in a more specific namespace
- I think in this directory we should only have "public" hooks that are meant to be included in some external front components as a catalog "vendor". Locale is not part of the catalog business.

Comment (tseho) [reply components/catalogs/front/src/hooks/useLocales.ts:2]: They have to share this hook between 2 majors components.
We have 2 solutions:
- we copy/paste the hook/model in both
- we move it to a shared namespace

Given the fact that locales are not really specific to either business, I think it's ok to move it here.

Comment (RiffVentura) [reply components/catalogs/front/src/hooks/useLocales.ts:2]: I'm leaning more towards moving the model and hook higher (we could also not export what we deem unnecessary through an index file)

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/front/src/hooks/useLocales.ts (hunk 1 above)
Comment (guillaumefer-akeneo) [reply components/catalogs/front/src/hooks/useLocales.ts]: Ok, I moved the Locale on the higher/shared models directory.

**Review by masacc — APPROVED** [review]

File: components/catalogs/front/src/hooks/useLocales.ts (hunk 1 above)
Comment (willy-ahva) [inline components/catalogs/front/src/hooks/useLocales.ts]: The Locale model must be moved to `src/model/` you'll see a channel.d.ts in the same folder.

**Review by willy-ahva — APPROVED** [review]

**Review by willy-ahva — APPROVED** [review]

**Pull request closed without merging**